source/common/common/assert.h

```diff
@@ -16,6 +16,21 @@ namespace Envoy {
     }                                                                                              \
   } while (false)
 
+/**
+ * A version of RELEASE_ASSERT that allows for ostream style extra data.
+ */
+#define VERBOSE_RELEASE_ASSERT(X, Y)                                                               \
```

Comment:
Should this just be the `RELEASE_ASSERT` implementation in the interest of economy of mechanism?

Reply:
FWIW, I've wanted something like this for a while, so I would be in favor of adding this.

Reply:
Cool.  If you're up for it, I'm all for doing the more invasive ostream style default-release-assert.  And possibly moving the other asserts over as well.

I think at that point I'll probably split it out of the test flake PR though :-)

Reply:
Any reason not to use the existing fmtlib-style for this?

Reply:
My weak argument is to be consistent with the ASSERT_EQ/EXPECT_EQ code in test but the real reason is I like ostream way better.  I believe that fmt is more consistent with the code base so alternately I could look into a variable 1/2 length macro with a fmt string as the second argument.

Reply:
I prefer consistency with the code base, but I don't feel strongly about it. Anyone else have an opinion on this?

Reply:
An argument in favor of ostream is that if we implement it that way, we can easily add the fmtlib variant as a wrapper on top, e.g. 
```
#define RELEASE_ASSERT(X, Y, ...) RELEASE_ASSERT(X) << fmt::format(Y, __VARARGSTUFF_)
```
so, doing it ostream provides maximum flexibility.

Reply:
No strong opinion from me. Whatever you all think is best. Happy to see this land though, there are many times in prod code when I think it would be nice to actually have a more detailed message of a RELEASE_ASSERT() (I would go as far as to say we should ultimately audit all uses and force everyone to use the new version). Agree this can be split out though...